src/docs/codelabs/layout-basics.md

```diff
@@ -1,427 +1,646 @@
+    Row(), // Second empty Row
+   ], 
+  ); // <--- Closing parentheses for the Column that wraps the Row
 
-If you run the code at this point, you'll see that the two `Text`
-widgets are placed right up against each other rather than at
-opposite ends of the `Row`, which isn't right.
-You can fix this by setting the `Row`'s `mainAxisAlignment`
-property to `spaceBetween`, which puts any extra space between
-the two `Text` widgets.
+  ```
 
-<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=e6e07bbe96255b762163cf3e40906944&split=60" width="100%" height="800px"></iframe>
-
-The last step is to get those icons in place at the bottom of the card:
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/c3ac34ed8952724a0ecb0af1445c2af8
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=c3ac34ed8952724a0ecb0af1445c2af8&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+#### Exercise: Enter contact information 
+{:.no_toc}
+{{site.alert.secondary}}
+  Enter two `Text` widgets inside the first empty `Row` : 
+
+<ul markdown="1">
+  <li markdown="1">  
+  The first `Text` widget contains the address `123 Main Street`. 
+  </li>
+  <li markdown="1">
+  The second `Text` widget contains the phone number `(415) 555-0198`.
+  </li>
+</ul>
+
+ For the first empty `Row`, 
+ set the `mainAxisAlignment` property to `MainAxisAlignment.SpaceBetween`.
 
-* Add four `Icon` widgets to the last `Row`'s list of
-  children. You can use whichever icon resources you
-  like, but these would be a good way to show that your
-  imaginary developer focuses on accessibility,
-  fast development, and multi-platform apps:
-  * `Icons.accessibility`
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/c5be61116652927c5d92262fce1b5360
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=c5be61116652927c5d92262fce1b5360&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+### Part 3
+{:.no_toc}
+#### Exercise: Add four icons
+{:.no_toc}
+{{site.alert.secondary}}
+  Enter the following `Icon` widgets inside the second empty `Row`:
+
+  * `Icons.accessibility` 
   * `Icons.timer`
   * `Icons.phone_android`
   * `Icons.phone_iphone`
 
-* Set the `Row`'s `mainAxisAlignment` property to
-  `MainAxisAlignment.spaceAround`
+  For the second empty `Row`, 
+  set the `mainAxisAlignment` property to `MainAxisAlignment.spaceAround`.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/dae36611fc9af04c4b9d0fbc3429275e
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=dae36611fc9af04c4b9d0fbc3429275e&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## What's next? 
+
+Congratulations, you finished this codelab! If you'd like to learn more about Flutter, here are couple of suggestions for resources worth exploring: 
+
```

Comment:
These resources seem fine, but I expected links to other layout resources, like /docs/development/ui/layout.